gradle/wrapper/gradle-wrapper.properties

```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0.2-bin.zip
 zipStoreBase=GRADLE_USER_HOME
```

Comment:
Please also add .gitattributes from #236 (this is to avoid line ending problems in .bat files and fix line endings in all other files to LF) and specify `distributionSha256Sum` property here.

Reply:
> this is to avoid line ending problems in .bat files and fix line endings in all other files to LF

Why are they problems? Does this cause any issues? I see value in sticking to the defaults unless this can be helped.

> and specify `distributionSha256Sum` property here.

I'd rather not. Without it, updating the Gradle version is as simple as `./gradlew wrapper --gradle-version 8.0.2`, but if we stick to adding `distributionSha256Sum`, this introduces a bit of friction that may be tiny but is still ultimately useless.

Reply:
> this introduces a bit of friction 

Still a good practice

> Why are they problems? Does this cause any issues?

I remember there were line ending differences in .bat files last time we updated the wrapper and this could make .bat non-functional on Windows, .gitattributes are intended to unify line ending handling on different platforms and not depend on contributors' autocrlf setting.

Reply:
> [w.r.t. .gitattributes]

Ok, thank you, that makes sense. Added.

> Still a good practice

I tried to check what the hash does manually, and I couldn't find any special behavior around it other than updating the wrapper itself. I intentionally wrote a wrong hash there, stopped all the Gradle daemons, and removed the `.gradle/` and `~/.gradle` directories, but after that, the build still proceeded as usual. So, what is this good practice good for?

Reply:
It checks the integrity of the Gradle dist when downloading it first time, if the check is failed, an error like this is prinited:
```
Downloading https://services.gradle.org/distributions/gradle-8.0.2-bin.zip
...........10%............20%............30%............40%............50%............60%...........70%............80%............90%............100%
Exception in thread "main" java.lang.RuntimeException: Verification of Gradle distribution failed!

Your Gradle distribution may have been tampered with.
Confirm that the 'distributionSha256Sum' property in your gradle-wrapper.properties file is correct and you are downloading the wrapper from a trusted source.

 Distribution Url: https://services.gradle.org/distributions/gradle-8.0.2-bin.zip
Download Location: C:\Users\...\.gradle\wrapper\dists\gradle-8.0.2-bin\25jlreiuz6u3xu2phlpa2vv4m\gradle-8.0.2-bin.zip
Expected checksum: 'f6b8596b10cce501591e92f229816aa4046424f3b24d771751b06779d58c8ec4'
  Actual checksum: 'ff7bf6a86f09b9b2c40bb8f48b25fc19cf2b2664fd1d220cd7ab833ec758d0d7'
```

Reply:
After thinking about it, I could imagine a scenario in which this SHA-256 could be helpful: given the recent wave of Github implementing automated checks across repositories (notifications about vulnerabilities, leaked keys, etc.), maybe Github will also notify about a hash mismatch.